### PR TITLE
Removing attribute cleanups.

### PR DESCRIPTION
**Pull request by timothy-king**
Code is in a draft state. If someone can "clean up" this commit, and some surrounding comments I'd appreciate it.

### CODE REVIEW

**Comment by 4tXJ7f** [comment]
Happy to take this over. Could you give me write [access](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/)?

**Comment by timothy-king** [comment]
I did a second pass. This should be good to review now.

File: src/expr/node_manager.h [hunk 1]
```diff
@@ -128,10 +128,7 @@ class NodeManager {
    * contexts, like as a key in attribute tables), even though
    * normally it's an error to have a TNode to a node value with a
    * reference count of 0.  Being "under deletion" also enables
-   * assertions that inc() is not called on it.  (A poorly-behaving
-   * attribute cleanup function could otherwise create a "Node" that
-   * points to the node value that is in the process of being deleted,
-   * springing it back to life.)
+   * assertions that inc() is not called on it.
    */
   expr::NodeValue* d_nodeUnderDeletion;
```
Comment (4tXJ7f) [inline src/expr/node_manager.h]: Do we still need to keep track of this?

Comment (timothy-king) [reply src/expr/node_manager.h]: Good question, but this is a bit out of scope.

**Pull request merged**